libs/core/algorithms/include/hpx/parallel/algorithms/count.hpp

```diff
@@ -64,17 +56,41 @@ namespace hpx {
     ///           is defined by \a std::iterator_traits<FwdIterB>::difference_type.
     ///           The \a count algorithm returns the number of elements
     ///           satisfying the given criteria.
+    template <typename ExPolicy, typename FwdIter, typename T>
+    typename util::detail::algorithm_result<ExPolicy,
+        typename std::iterator_traits<FwdIter>::difference_type>::type
+    count(ExPolicy&& policy, FwdIter first, FwdIter last, T const& value);
+    
+    /// Returns the number of elements in the range [first, last) satisfying
+    /// a specific criteria. This version counts the elements that are equal to
+    /// the given \a value.
+    ///
+    /// \note   Complexity: Performs exactly \a last - \a first comparisons.
+    /// \tparam InIter      The type of the source iterator used (deduced).
+    ///                     This iterator type must meet the requirements of an
+    ///                     input iterator.
+    /// \tparam T           The type of the value to search for (deduced).
     ///
-    template <typename ExPolicy, typename FwdIterB, typename FwdIterE,
-        typename T, typename Proj = util::projection_identity>
+    /// \param first        Refers to the beginning of the sequence of elements
+    ///                     the algorithm will be applied to.
+    /// \param last         Refers to the end of the sequence of elements the
+    ///                     algorithm will be applied to.
+    /// \param value        The value to search for.
+    ///
+    ///
+    /// \returns  The \a count algorithm returns a \a difference_type (where \a difference_type
+    ///           is defined by \a std::iterator_traits<InIter>::difference_type.
+    ///           The \a count algorithm returns the number of elements
+    ///           satisfying the given criteria.
+    ///
+    template <typename InIter, typename T>
     typename util::detail::algorithm_result<ExPolicy,
```

Comment:
Synchronous algorithms (the ones not taking an execution policy) never return the `algorithm_result_t<>` helper type, but always the actual return type (here `typename std::iterator_traits<InIter>::difference_type`).

Reply:
Yes, missed this somehow. Check now?